src/app/tests/TestPowerSourceCluster.cpp

```diff
@@ -34,6 +35,18 @@
 
 #include <vector>
 
+namespace {
+chip::EndpointId numEndpoints = 0;
+}
+extern uint16_t emberAfGetClusterServerEndpointIndex(chip::EndpointId endpoint, chip::ClusterId cluster, uint16_t fixedClusterServerEndpointCount) {
```

Comment:
is "extern" needed here?

Reply:
yup - it needs to be exported because the test is defining it, not the base. Basically, the test uses this mock attribute server instead of the normal one and the mock doesn't define this function. I didn't want to add this in the mock, because this is really quite specific to this test and if tests are using it, they probably want to define it themselves.